src/execution_plan/ops/shared/create_functions.c

```diff
@@ -165,7 +165,10 @@ PendingProperties *ConvertPropertyMap(Record r, const PropertyMap *map) {
 		converted->values[i] = AR_EXP_Evaluate(map->values[i], r);
 		// Emit an error and return NULL if we're trying to add an invalid type.
 		if(!(SI_TYPE(converted->values[i]) & SI_VALID_PROPERTY_VALUE)) {
+			converted->property_count = i;
+			PendingPropertiesFree(converted);
 			Error_InvalidPropertyValue();
+			QueryCtx_RaiseRuntimeException();
 			break;
```

Comment:
Is this block covered by a test? I can see that MERGE is covered but I'm not sure about CREATE.